internal/pkg/docker/dockerengine/dockerengine.go

```diff
@@ -201,6 +210,72 @@ func (c DockerCmdClient) Push(ctx context.Context, uri string, w io.Writer, tags
 	return parts[1], nil
 }
 
+func (in *RunOptions) generateRunArguments() []string {
+	args := []string{"run"}
+
+	//Add container name option.
+	if in.ContainerName != "" {
+		args = append(args, "--name", in.ContainerName)
+	}
+
+	// Add port mappings.
+	for hostPort, containerPort := range in.ContainerPorts {
+		args = append(args, "-p", fmt.Sprintf("%s:%s", hostPort, containerPort))
+	}
+
+	// Add network option if it's not a "pause" container.
+	if in.ContainerName != "pause" {
+		args = append(args, "--network", "container:pause")
+	}
+
+	//Add secrets as environment variables.
+	for key, value := range in.Secrets {
+		args = append(args, "-e", fmt.Sprintf("%s=%s", key, value))
+	}
+
+	//Add environment variables.
+	for key, value := range in.EnvVars {
+		args = append(args, "-e", fmt.Sprintf("%s=%s", key, value))
+	}
+
+	//Add the image name.
+	if in.ImageURI != "" {
+		args = append(args, in.ImageURI)
+	}
+
+	//Add "sleep infinity" command for the "pause" container
+	if in.ContainerName == "pause" {
+		args = append(args, "sleep", "infinity")
+	}
+	return args
+}
+
+// Run runs a Docker container with the sepcified options.
+func (c DockerCmdClient) Run(ctx context.Context, options *RunOptions) error {
+	args := options.generateRunArguments()
+
+	//Execute the Docker run command.
+	if err := c.runner.RunWithContext(ctx, "docker", args); err != nil {
+		return fmt.Errorf("running container: %w", err)
+	}
+	return nil
+}
+
+// IsContainerRunning checks if a specific Docker container is running.
+func (c DockerCmdClient) IsContainerRunning(containerName string) (bool, error) {
+	buf := &bytes.Buffer{}
+	err := c.runner.Run("docker", []string{"ps", "-q", "-f", "name=" + containerName}, exec.Stdout(buf))
+	if err != nil {
+		return false, fmt.Errorf("run docker ps: %w", err)
+	}
+
+	output := strings.TrimSpace(buf.String())
+	if output == "" {
+		return false, nil // The container is not running.
+	}
+	return true, nil // The container is running.
```

Comment:
doesn't this return true if _any_ containers are running? it doesn't seem like we are filtering to ensure `containerName` is running

Reply:
It does show only if the container with specific name is running the -f flag which is used is to filter right? I will change it to --filter

Reply:
oh, cool! `--filter` will help - i thought `-f` was just for the output format 🤦‍♂️